cmd/kube-apiserver/app/server.go

```diff
@@ -279,6 +280,23 @@ func Run(s *options.APIServer) error {
 	}
 	storageDestinations.AddAPIGroup("", etcdStorage)
 
+	// autoscaling group
+	if !apiGroupVersionOverrides["autoscaling/v1"].Disable {
+		autoGroup, err := registered.Group(autoscaling.GroupName)
+		if err != nil {
+			glog.Fatalf("Autoscaling API is enabled in runtime config, but not enabled in the environment variable KUBE_API_VERSIONS. Error: %v", err)
+		}
+		if _, found := storageVersions[autoGroup.GroupVersion.Group]; !found {
+			glog.Fatalf("Couldn't find the storage version for group: %q in storageVersions: %v", autoGroup.GroupVersion.Group, storageVersions)
+		}
+		autoscalingEtcdStorage, err := newEtcd(s.EtcdServerList, api.Codecs, storageVersions[autoGroup.GroupVersion.Group], s.EtcdPathPrefix, s.EtcdQuorumRead)
+		if err != nil {
+			glog.Fatalf("Invalid autoscaling storage version or misconfigured etcd: %v", err)
+		}
+		storageDestinations.AddAPIGroup(autoscaling.GroupName, autoscalingEtcdStorage)
```

Comment:
I am really skeptical that this creates a codec that reads & writes the correct types.

Can you add something like this change to verify? https://github.com/lavalamp/kubernetes/commit/6ecf1a33a220c3ca095acf7d5f51228a328c7cef